### PR TITLE
[jest] fix `jest.useFakeTimers()` config typings

### PR DESCRIPTION
**Pull request by mrazauskas**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: see Jest repo [here](https://github.com/facebook/jest/blob/e21c5aba950f6019bbfde2f8233ac96d1fcaef42/packages/jest-types/src/Config.ts#L42-L94) and [here](https://github.com/facebook/jest/blob/e21c5aba950f6019bbfde2f8233ac96d1fcaef42/packages/jest-environment/src/index.ts#L332-L334)
---

### Summary

As you know, Jest implements legacy and Sinon-based fake timers. To use legacy timers user must call `jest.useFakeTimers({ legacyFakeTimers: true })`. The legacy timers **do not take any additional options**.

If the Sinon-based timers are used, the user **can pass more options**, e.g.: `jest.useFakeTimers({ timerLimit: 50 })`.

Current typings allow passing options for both implementations of fake timers. This PR is fixing this issue so that only Sinon-based timers would be allowed to take options. The code and TSDoc comments are simply copied from Jest repo [here](https://github.com/facebook/jest/blob/e21c5aba950f6019bbfde2f8233ac96d1fcaef42/packages/jest-types/src/Config.ts#L42-L94) and [here](https://github.com/facebook/jest/blob/e21c5aba950f6019bbfde2f8233ac96d1fcaef42/packages/jest-environment/src/index.ts#L332-L334).

### CODE REVIEW

**Comment by regevbr** [comment]
@mrazauskas would you mind removing me from the maintainers list?

File: types/jest/index.d.ts [hunk 1]
```diff
@@ -24,7 +24,6 @@
 //                 Tony Hallett <https://github.com/tonyhallett>
 //                 Jason Yu <https://github.com/ycmjason>
 //                 Pawel Fajfer <https://github.com/pawfa>
-//                 Regev Brody <https://github.com/regevbr>
```
Comment (mrazauskas) [inline types/jest/index.d.ts:27]: @regevbr as per your request: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62468#issuecomment-1262119436

Comment (regevbr) [reply types/jest/index.d.ts:27]: thanks!!!

**Review by r3nya — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

**Comment by mrazauskas** [comment]
Ready to merge

**Pull request merged**